### DISCUSSION TITLE
How to Display a Splash Window Before the Main Window in WinUI3 C++

### DISCUSSION THREAD

**Discussion by toto45638** [microsoft/microsoft-ui-xaml#10037] [Q&A] [answered]
I am currently trying to display a splash window using WinUI3 in C++. In the App.xaml.cpp, I implemented the splash window as shown below, but it appears completely white until the main window is displayed.
The contents of the splash window are displayed at the same time the main window is rendered.

How can I display the contents of the splash window before the main window shows up?

SampleApp::SplashScreen is a Window.

``` C++
void App::OnLaunched([[maybe_unused]] LaunchActivatedEventArgs const& e)
{
    SapmleApp::SplashScreen splash_screen;
    splash_screen.Activate();
    
    // Sleep 5 seconds
    std::this_thread::sleep_for(std::chrono::seconds(5));
    
    window = make<MainWindow>();
    window.Activate();

    splash_screen.Close();
}

**Comment by whiskhub** [comment, marked as answer]
Probably because you are blocking the main thread with the "sleep for 5 seconds". Try creating a `DispatcherQueueTimer` with 5 seconds and then asynchronously show the main window once it elapses.

**Reply by toto45638** [reply]
The splash screen was displayed! Thank you very much.
However, I noticed that the WinUI3 window initially appears white, which seems to be a feature. It’s a bit unusual. It looks like I still have some work to do.

Thanks for your help!

**Reply by DarranRowe** [reply]
I would have to capture the desktop with a higher refresh video than I normally do, but it didn't look like the window was initially white for more than 100ms for some of my test projects.
How complex are your .xaml files, and do you do a lot of work on the main thread? If you are, then you really should be pushing work off onto the thread pool. C++/WinRT does enable the use of coroutines for this kind of work.

**Reply by toto45638** [reply]
The file is for the splash window, which simply displays an image. I don’t mind other windows being white, but the splash window bothers me a bit. For now, I made the window transparent for the first 100ms using a coroutine, and that eliminated the discomfort. Thank you! If you have any other suitable methods, I would appreciate your guidance.

**Reply by whiskhub** [reply]
In the WinUIEx repo, a splash screen is currently being implemented. You may want to check it out as well. https://github.com/dotMorten/WinUIEx/pull/186

**Reply by toto45638** [reply]
Thanks for the info. Unfortunately, I use C++, so I can't use that extension, but I will refer to it. it's a little disappointing that there is no extension for WinUI in C++.
